### PR TITLE
fix formatting in depthfirst search and add examples for friendship g…

### PR DESCRIPTION
**Pull request by srv-twry**
- Fixed a few formatting mistakes
- Added examples for the friendship graph generator

### CODE REVIEW

File: src/Traversals/depthfirst.jl [hunk 1]
```diff
@@ -2,13 +2,11 @@ import Base.showerror
...
 Base.showerror(io::IO, e::CycleError) = print(io, "Cycles are not allowed in this function.")
 
-
-
 struct DepthFirst{F<:Function} <: TraversalAlgorithm
     neighborfn::F
 end
 
-DepthFirst(;neighborfn=outneighbors) = DepthFirst(neighborfn)
+DepthFirst(;neighborfn = outneighbors) = DepthFirst(neighborfn)
```
Comment (sbromberger) [inline src/Traversals/depthfirst.jl]: we don't put spaces between assignments in signatures (this is the exception to the general rule).

File: src/Traversals/depthfirst.jl [hunk 2]
```diff
@@ -139,7 +138,7 @@ function postvisitfn!(s::CycleState{T}, u) where T
     return true
 end
 
-@traitfn function is_cyclic(g::AG::IsDirected, alg::TraversalAlgorithm=DepthFirst()) where {T, AG<:AbstractGraph{T}}
+@traitfn function is_cyclic(g::AG::IsDirected, alg::TraversalAlgorithm = DepthFirst()) where {T, AG<:AbstractGraph{T}}
```
Comment (sbromberger) [inline src/Traversals/depthfirst.jl]: see above

File: src/Traversals/depthfirst.jl (hunk 1 above)
Comment (srv-twry) [reply src/Traversals/depthfirst.jl]: done. thanks

**Comment by sbromberger** [comment]
Thank you, @srv-twry !

**Pull request merged**